mars/tiles.py

```diff
@@ -54,7 +54,7 @@ def _inplace_tile(self):
     def build_graph(self, graph=None, cls=DAG, tiled=False, compose=True,
                     **build_chunk_graph_kwargs):
         tileable_graph = graph if not tiled else None
-        tileable_graph_builder = TileableGraphBuilder(graph=tileable_graph, graph_cls=cls)
+        tileable_graph_builder = get_tileable_graph_builer()(graph=tileable_graph, graph_cls=cls)
```

Comment:
Add an optimize option here and choose right graph builder according to the option.